pkg/asset/installconfig/ibmcloud/metadata.go

```diff
@@ -90,3 +92,8 @@ func (m *Metadata) Client() (*Client, error) {
 	}
 	return m.client, nil
 }
+
+// NewIamAuthenticator returns a new IamAuthenticator for using IBM Cloud services.
+func NewIamAuthenticator(apiKey string) (*core.IamAuthenticator, error) {
```

Comment:
nit: This function does not need to be public. It is only used in the pkg/asset/installconfig/ibmcloud package.

Reply:
okay, I'll change it to private then

Reply:
Changed back to `public` for use during `destroy` as well